packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -910,24 +1030,23 @@ class Actions extends StatefulWidget {
   ) {
     assert(intent != null);
     assert(context != null);
-    Action<T>? action;
-    InheritedElement? actionElement;
+    Object? returnValue;
 
-    _visitActionsAncestors(context, (InheritedElement element) {
+    final bool actionFound = _visitActionsAncestors(context, (InheritedElement element) {
       final _ActionsMarker actions = element.widget as _ActionsMarker;
-      final Action<T>? result = actions.actions[intent.runtimeType] as Action<T>?;
-      if (result != null) {
-        actionElement = element;
```

Comment:
According to the documentation this should throw even when a disabled action is found:

>  /// This method will throw an exception if no ambient [Actions] widget is
  /// found, or if the given `intent` doesn't map to an enabled action in any of
  /// the [Actions.actions] maps that are found.

Currently it doesn't